linux: disable NTFS_FS, enable NTFS3_LZX_XPRESS and NTFS3_FS_POSIX_ACL

SuperSamus opened the pull request:
###### Description of changes

Disables the old read-only NTFS driver, and enable the [recommended options](https://github.com/torvalds/linux/blob/master/fs/ntfs3/Kconfig) for the new NTFS3 driver.
[Arch](https://github.com/archlinux/svntogit-packages/blob/packages/linux/trunk/config#L9959) and [Fedora](https://gitlab.com/cki-project/kernel-ark/-/tree/os-build/redhat/configs/fedora/generic) use the same configuration.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

ulrikstrid committed:
ocamlPackages.pbkdf: 1.1.0 -> 1.2.0

sarcasticadmin committed:
init conserver 8.2.7

ElvishJerricco committed:
shutdown: Protect system from make-initrd-ng

SuperSamus committed:
linux: disable NTFS_FS, enable NTFS3_LZX_XPRESS and NTFS3_FS_POSIX_ACL

fortuneteller2k commented:
Please target the `staging` branch.

lovesegfault commented:
> Please target the `staging` branch.

You don't need staging for changes to the kernel.

lovesegfault commented on pkgs/os-specific/linux/kernel/common-config.nix:
```diff
@@ -394,6 +394,10 @@ let
       EXT4_FS_SECURITY  = yes;
       EXT4_ENCRYPTION   = option yes;
 
+      NTFS_FS = no;
```
```suggestion
      NTFS_FS = no;
      NTFS3_FS = module;
```

SuperSamus replied:
It's already implied for `NTFS3_FS` to be a module (same for every other filesystem there).

Atemu commented on pkgs/os-specific/linux/kernel/common-config.nix:
```diff
@@ -394,6 +394,10 @@ let
       EXT4_FS_SECURITY  = yes;
       EXT4_ENCRYPTION   = option yes;
 
+      NTFS_FS = no;
+      NTFS3_LZX_XPRESS = yes;
+      NTFS3_FS_POSIX_ACL = yes;
```
Older kernels don't have this option, do they?

stehessel committed:
cbfmt: 0.1.1 -> 0.1.4

SuperSamus commented:
Is there anything else to do for this to be merged?

r-ryantm committed:
picard: 2.8.1 -> 2.8.3

smasher164 committed:
vscode-extensions.azdavis.millet: init at 0.3.5

jakubgs committed:
go-ethereum: add support for Auth RPC CLI flags

The split of Ethereum into Execution Layer and Consensus Layer adds a
requirement for communication between execution client and consensus
client using secur JWT tokens. In Geth this is configurable using the
`--authrpc.*` CLI flags which are currently not exposed by this service.

For more details read the following article:
https://geth.ethereum.org/docs/interface/consensus-clients

Signed-off-by: Jakub Sokołowski <jakub@status.im>

ajs124 committed:
mariadb_109: init

https://mariadb.com/kb/en/mariadb-1092-release-notes/

sinuscosinustan committed:
changelogger: 0.5.2 -> 0.5.3

newAM committed:
github-runner: 2.295.0 -> 2.296.0

parthy committed:
ipxe: Allow to specify additional options

Overriding the `enabledOptions` attribute only works if `configurePhase`
is also overridden, which makes it a bit inconvenient.

ajs124 committed:
xdaliclock: 2.44 -> 2.45

https://www.jwz.org/blog/2022/08/dali-clock-2-45-released/

superherointj committed:
asus-ec-sensors: unstable-2021-12-16 -> unstable-2022-07-10

superherointj committed:
nautilus-open-any-terminal: fix build

SuperSandro2000 committed:
python310Packages.flask: add key reverse dependencies

fabaff committed:
python310Packages.accuweather: 0.3.0 -> 0.4.0

fabaff committed:
python310Packages.aioaladdinconnect: 0.1.42 -> 0.1.43

fabaff committed:
Merge pull request #188378 from fabaff/aioaladdinconnect-bump

python310Packages.aioaladdinconnect: 0.1.42 -> 0.1.43

fabaff committed:
Merge pull request #188376 from fabaff/accuweather-bump

python310Packages.accuweather: 0.3.0 -> 0.4.0

qbit committed:
yash: 2.52 -> 2.53

Upstream changes: https://github.com/magicant/yash/blob/trunk/NEWS

superherointj committed:
Merge pull request #180725 from ulrikstrid/ulrikstrid--pbkdf-1_2_0

ocamlPackages.pbkdf: 1.1.0 -> 1.2.0

bobby285271 committed:
Merge pull request #188029 from tomsiewert/update-changelogger

changelogger: 0.5.2 -> 0.5.3

superherointj committed:
Merge pull request #187819 from smasher164/millet-vscode-extension

vscode-extensions.azdavis.millet: init at 0.3.5

tfc committed:
Merge pull request #188292 from parthy/ipxe-additional-options

ipxe: Allow to specify additional options

bobby285271 committed:
Merge pull request #188034 from stehessel/update-cbfmt-0.1.4

cbfmt: 0.1.1 -> 0.1.4

Commit:
wpsoffice: 11.1.0.9615 -> 11.1.0.11664

SuperSandro2000 committed:
python310Packages.click: add typer to passthru.tests

SuperSandro2000 committed:
python310Packages.werkzeug: add moto, sentry-sdk to passthru.tests

mweinelt committed:
Merge pull request #188397 from SuperSandro2000/python-passthru-tests

mweinelt committed:
Merge pull request #188377 from SuperSandro2000/flask-passthru-tests

pennae committed:
stdenv: remove deprecated adapters

dasJ committed:
Merge pull request #188026 from helsinki-systems/init/mariadb109

mariadb_109: init

bobby285271 committed:
Merge pull request #178289 from wineee/wps

wpsoffice: 11.1.0.9615 -> 11.1.0.11664

bachp committed:
Merge pull request #187876 from jakubgs/geth-add-authrpc

go-ethereum: add support for Auth RPC CLI flags

lovesegfault commented:
Other than the rebase, no, I can merge it once it's done.

msfjarvis committed:
nerdfonts: 2.1.0 -> 2.2.0

superherointj committed:
Merge pull request #188388 from helsinki-systems/upd/xdaliclock

xdaliclock: 2.44 -> 2.45

superherointj committed:
Merge pull request #188407 from msfjarvis/hs/nerdfonts-2.0.0

nerdfonts: 2.1.0 -> 2.2.0

ElvishJerricco committed:
Merge pull request #185085 from ElvishJerricco/shutdown-ramfs-protection

shutdown: Protect system from make-initrd-ng

Artturin committed:
Merge pull request #188400 from pennae/remove-deprecated-adapters

Artturin committed:
Merge pull request #188350 from superherointj/broken-nautilus-open-any-terminal

aanderse committed:
Merge pull request #188117 from newAM/github-runner

github-runner: 2.295.0 -> 2.296.0

superherointj committed:
Merge pull request #188341 from superherointj/package-asus-ec-sensors-unstable-2022-07-10

asus-ec-sensors: unstable-2021-12-16 -> unstable-2022-07-10

onny committed:
Merge pull request #180679 from r-ryantm/auto-update/picard

picard: 2.8.1 -> 2.8.3

lovesegfault commented:
Closing to avoid spamming everyone. Please reopen this PR targeting master, and not staging, as is customary for kernel work.
